Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm

```diff
@@ -171,6 +171,10 @@ static EGLDisplay initializeEGLDisplay(const GraphicsContextGLAttributes& attrs)
     , m_resourceOwner(WTFMove(resourceOwner))
     , m_drawingBufferColorSpace(DestinationColorSpace::SRGB())
 {
+    // Set the max internal framebuffer size.
+    int minDimension = std::min(MAX_TEXTURE_SIZE, std::min(MAX_VIEWPORT_DIMS, MAX_RENDERBUFFER_SIZE));
```

Comment:
grep the webkit whole project to see how these are queried..

Reply:
These should be queried  in ANGLE code after the context has been created.

The platform specific part for cocoa should then take the min of this and iosurface max size..